prime-router/docs/design/design/data-model.md

```diff
@@ -0,0 +1,355 @@
+
+In ReportStream, Reports flow from a "sender" to one or more "receivers", both of which belong to exactly one ReportStream "Organization". An Organization can have one or more ReportStream Receivers and one or more ReportStream Senders under it. Due to the nature of the use case mostly commonly supported by ReportStream (ELR), usually, but not always, an Organization either has only senders or only receivers. If an Organization contains only senders, the Organization itself is colloquially referred to as a "Sender." If an Organization contains only receivers, the Organization itself is colloquially referred to as a "Receiver."
+
+For example, in the case of California, there is an Organization called _ca-dph_, with the following Receivers (some omitted for brevity) and no Senders.
+
+- elr-saphirestage
+- elr-test
+
+The ReportStream team conversationally refers to this organization as "California Department of Public Health, one of our receivers" but technically speaking it is more accurate and descriptive to say "California Department of Public Health, a ReportStream Organization with multiple Receivers and no Senders."
+
+#### Associated Tables
+
+- setting 
+
+#### `setting` table
+
+The `setting` table contains three types of settings, indicated by the `type` column: `ORGANIZATION`, `SENDER`, `RECEIVER`. Each row represents one of these objects, with the `values` json column containing the settings for that object. 
+
+Each row also has an `organization_id`, which is always null for an `ORGANIZATION` type but, in the case of `SENDER` and `RECEIVER` types, links to the particular `ORGANIZATION` type that Sender or Receiver belongs to. This is how the relationship between Organizations and Senders/Receivers is modeled in the database.
+
+Furthermore, the `setting` table also supports versioning. When a change is made to a setting, the `values` column does not get overriden, instead, a new row is created and the `version` column is incremented for the new row, as well as the `is_active` column being updated accordingly for both the new and old row.
+
+**Example Table** (`created_at` and `created_by` columns redacted for brevity).
+
+| setting_id | type         | name                | organization_id | values  | is_deleted | is_active | version |
+|------------|--------------|---------------------|-----------------|---------|------------|-----------|---------|
+| 229        | ORGANIZATION | flexion             | <null>          | { ... } | false      | true      | 3       |
+| 62         | RECEIVER     | simulated-lab       | 229             | { ... } | false      | true      | 8       |
+
+#### Organization Settings
+
+Organizations serve as containers for Senders and Receivers, but have their own settings as well. The settings for an Organization are self-explanatory. For more information on the `keys` parameter, see [SMART on FHIR oauth implementation](../features/0002-authentication.md#via-smart-on-fhir-oauth-implementation-details).
```

Comment:
Bit confusing. You say they are self explanatory but then refer to the keys param. Is this one that isn't? Is this something else? A transition would be good here.

Reply:
Updated! Let me know if still not clear and I can take another pass at it!

Reply:
Perfect!